File: src/java/magma/app/compile/value/Not.java
package magma.app.compile.value;

import magma.api.option.Option;
import magma.api.option.Some;
import magma.app.compile.CompileState;
import magma.app.compile.node.Node;
import magma.app.compile.type.Primitives;

public record Not(String child) implements Node {
    public String generate() {
        return this.child;
    }

    public Option<Node> toNode() {
        return new Some<Node>(this);
    }

    public Node resolve(CompileState state) {
        return Primitives.UNKNOWN;
    }


    public boolean is(String type) {
        return false;
    }
}

File: src/ts/magma/app/compile/value/Not.ts
import { Node } from "../../../../magma/app/compile/node/Node";
import { Option } from "../../../../magma/api/option/Option";
import { Some } from "../../../../magma/api/option/Some";
import { Type } from "../../../../magma/app/compile/type/Type";
import { CompileState } from "../../../../magma/app/compile/CompileState";
import { PrimitiveType } from "../../../../magma/app/compile/type/PrimitiveType";
export class Not implements Node {
	child: string;
	constructor (child: string) {
		this.child = child;
	}
	generate(): string {
		return this.child/*unknown*/;
	}
	toNode(): Option<Node> {
		return new Some<Node>(this)/*unknown*/;
	}
	resolve(state: CompileState): Type {
		return PrimitiveType.Unknown/*unknown*/;
	}
	is(type: string): boolean {
		return false/*unknown*/;
	}
}
